src/mlmc/sampling_pool.py

```diff
@@ -227,6 +228,9 @@ def _process_result(self, sample_id, result, err_msg, running_time, level_sim):
         if not err_msg:
             self._queues.setdefault(level_sim._level_id, queue.Queue()).put((sample_id, (result[0], result[1])))
             if not self._debug:
+                if int(sample_id[-7:]) < SamplingPool.N_SUCCESSFUL:
```

Comment:
Make the number of saved samples a class constant, e.g. SamplingPool.n_debug_samples.

Reply:
It is already class constant `SamplingPool.N_SUCCESSFUL`. 'sample_id[-7:]` gets digits representing sample id.